pkg/ibmvpc/naclAnalysis_test.go

```diff
@@ -46,112 +51,303 @@ func testSingleNACL(nacl *commonvpc.NACL) {
-		src           []string
-		dst           []string
-		expectedConns []*netset.TransportSet
-	}{
-		{
-			testName:      "a",
-			naclRules:     rulesTest1,
-			src:           []string{"1.1.1.1/32", "1.2.3.4/32", "1.2.3.4/32"},
-			dst:           []string{"10.0.0.0/24", "10.0.0.1/32", "10.0.0.0/32"},
-			expectedConns: []*netset.TransportSet{netset.AllTransports(), netset.NoTransports(), netset.AllTransports()},
-		},
+func equalKeys(first, second map[string]*commonvpc.ConnectivityResult) bool {
+	if len(first) != len(second) {
+		return false
 	}
+	keys1 := slices.Collect(maps.Keys(first))
+	sort.Strings(keys1)
+	keys2 := slices.Collect(maps.Keys(second))
+	sort.Strings(keys2)
+	// compare the concatenation result to validate equality of keys sets
+	return reflect.DeepEqual(keys1, keys2)
+}
 
-	for _, tt := range tests {
-		require.Equal(t, len(tt.src), len(tt.dst))
-		require.Equal(t, len(tt.src), len(tt.expectedConns))
-		for i := range tt.src {
-			src := newIPBlockFromCIDROrAddressWithoutValidation(tt.src[i])
-			dst := newIPBlockFromCIDROrAddressWithoutValidation(tt.dst[i])
-			disjointPeers := []*netset.IPBlock{dst}
-			expectedConn := tt.expectedConns[i]
-			res, _, _, _ := commonvpc.GetAllowedXgressConnections(tt.naclRules, src, dst, disjointPeers, true)
-			dstStr := dst.ToIPRanges()
-			actualConn := res[dstStr]
-			require.True(t, expectedConn.Equal(actualConn))
-		}
+func equalConnectivityMap(connectivityMap, other map[string]*commonvpc.ConnectivityResult) bool {
+	if !equalKeys(connectivityMap, other) {
+		return false
 	}
+	for ipBlockString, connectivityResult := range connectivityMap {
+		fmt.Printf("ipBlockString: %v\n", ipBlockString)
+		fmt.Printf("connectivityResult.AllowedConns: \n")
+		for ipRange, conn := range connectivityResult.AllowedConns {
+			fmt.Printf("ipRange.ToIPRanges(): %v\n", ipRange.ToIPRanges())
+			fmt.Printf("conn: %v\n", conn)
+		}
 
-	//nolint:all
-	/*src := common.FromCidr("1.1.1.1/32")
-	dst := common.FromCidr("10.0.0.0/24")
-	disjointPeers := []*netset.IPBlock{dst}
-	res := commonvpc.GetAllowedXgressConnections(rulesTest1, src, dst, disjointPeers, true)
-	for d, c := range res {
-		fmt.Printf("%s => %s : %s\n", src.ToIPAdress(), d, c.String())
-		require.True(t, c.Equal(getAllConnSet()))
-	}*/
+		fmt.Printf("connectivityResult.AllowRules: \n")
+		for ipRange, conn := range connectivityResult.AllowRules {
+			fmt.Printf("ipRange.ToIPRanges(): %v\n", ipRange.ToIPRanges())
+			fmt.Printf("conn: %v\n", conn)
+		}
+		fmt.Printf("connectivityResult.DeniedConns: \n")
+		for ipRange, conn := range connectivityResult.DeniedConns {
+			fmt.Printf("ipRange.ToIPRanges(): %v\n", ipRange.ToIPRanges())
+			fmt.Printf("conn: %v\n", conn)
+		}
 
-	fmt.Printf("done\n")
+		fmt.Printf("connectivityResult.DenyRules: \n")
+		for ipRange, conn := range connectivityResult.DenyRules {
+			fmt.Printf("ipRange.ToIPRanges(): %v\n", ipRange.ToIPRanges())
+			fmt.Printf("conn: %v\n", conn)
+		}
```

Comment:
Do we need all these prints?
We already have huge logs for our tests - it makes it hard to search for failures.
So unless there is a failure, I prefer tests to produce only very little text if at all.

Reply:
not needed
removed

Reply:
in https://github.com/np-guard/vpc-network-config-analyzer/pull/933